### PR TITLE
Minor C code generation/utility fixes

### PR DESCRIPTION
**Pull request by dalcinl**

### CODE REVIEW

File: Cython/Compiler/ModuleNode.py [hunk 1]
```diff
@@ -358,7 +358,9 @@ def generate_c_code(self, env, options, result):
 
         code = globalstate['before_global_var']
         code.putln('#define __Pyx_MODULE_NAME "%s"' % self.full_module_name)
-        code.putln("int %s%s = 0;" % (Naming.module_is_main, self.full_module_name.replace('.', '__')))
+        module_is_main = "%s%s" % (Naming.module_is_main, self.full_module_name.replace('.', '__'))
+        code.putln("extern int %s;" % module_is_main)
+        code.putln("int %s = 0;" % module_is_main)
```
Comment (scoder) [inline Cython/Compiler/ModuleNode.py]: I don't understand this. Why is it both `extern` and internal? And why does it need to be `extern` at all? Shouldn't it be `static` instead?

**Comment by scoder** [comment]
32e305ae01738d8a925248b9d0f4a08e187ce45c looks good to me.

File: Cython/Compiler/ModuleNode.py (hunk 1 above)
Comment (dalcinl) [reply Cython/Compiler/ModuleNode.py]: @scoder Maybe is not really required, but look at `bin/cython_freeze`. Are you 100% sure that making it `static` will not break things?

Comment (dalcinl) [reply Cython/Compiler/ModuleNode.py]: @scoder I can definitely confirm that `static` breaks `Demos/freeze`

```diff
[dalcinl@kl-13999 cython]$ git diff
diff --git a/Cython/Compiler/ModuleNode.py b/Cython/Compiler/ModuleNode.py
index 3c04f0838..e559cd43a 100644
--- a/Cython/Compiler/ModuleNode.py
+++ b/Cython/Compiler/ModuleNode.py
@@ -358,7 +358,7 @@ class ModuleNode(Nodes.Node, Nodes.BlockNode):
 
         code = globalstate['before_global_var']
         code.putln('#define __Pyx_MODULE_NAME "%s"' % self.full_module_name)
-        code.putln("int %s%s = 0;" % (Naming.module_is_main, self.full_module_name.replace('.', '__')))
+        code.putln("static int %s%s = 0;" % (Naming.module_is_main, self.full_module_name.replace('.', '__')))
         code.putln("")
         code.putln("/* Implementation of '%s' */" % env.qualified_name)
```

```
[dalcinl@kl-13999 cython]$ PYTHONPATH=$PWD make -C Demos/freeze
../../bin/cython_freeze combinatorics lcmath > nCr.c
gcc -fPIC -fno-strict-aliasing -g -O2 -Wall -Wextra -I/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/include/python2.7  -c -o nCr.o nCr.c
../../bin/cython  combinatorics.pyx
gcc -fPIC -fno-strict-aliasing -g -O2 -Wall -Wextra -I/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/include/python2.7  -c -o combinatorics.o combinatorics.c
../../bin/cython  lcmath.pyx
gcc -fPIC -fno-strict-aliasing -g -O2 -Wall -Wextra -I/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/include/python2.7  -c -o lcmath.o lcmath.c
gcc -u _PyMac_Error /usr/local/opt/python/Frameworks/Python.framework/Versions/2.7/Python -L/usr/local/opt/python/Frameworks/Python.framework/Versions/2.7/lib/python2.7/config  nCr.o combinatorics.o lcmath.o  -L/usr/local/opt/python/Frameworks/Python.framework/Versions/2.7/lib -L/usr/local/opt/python/Frameworks/Python.framework/Versions/2.7/lib/python2.7/config -lpython2.7 -o nCr
Undefined symbols for architecture x86_64:
  "___pyx_module_is_main_combinatorics", referenced from:
      _main in nCr.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [nCr] Error 1
```

**Comment by dalcinl** [comment]
@scoder To be honest, I HATE defining and using extern global vars this way. Furthermore, we should be using `extern "C"` in C++, otherwise we cannot mix C and C++ Cython-generated code to make a frozen binary. However, this fix if for the release branch, I don't want to mess too much with things to the point of introducing  regressions in third-party code. Let it be, Let it be...

Unless I hear further concerns from you, I'll merge this PR.

**Pull request merged**